src/mscorlib/src/System/Threading/Thread.cs

```diff
@@ -539,7 +552,22 @@ public void Abort()
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         private extern void AbortInternal();
 
-#if !FEATURE_CORECLR
+#if FEATURE_CORECLR
+        /*=========================================================================
+        ** Interrupts a thread that is inside a Wait(), Sleep() or Join().  If that
+        ** thread is not currently blocked in that manner, it will be interrupted
+        ** when it next begins to block.
+        =========================================================================*/
+        [System.Security.SecuritySafeCritical]  // auto-generated
+#pragma warning disable 618 // obsolete types: SecurityPermissionAttribute, SecurityAction
+        [SecurityPermission(SecurityAction.Demand, ControlThread = true)]
+#pragma warning restore 618 // obsolete types: SecurityPermissionAttribute, SecurityAction
+        public void Interrupt()
+        {
+            // TODO: implement
```

Comment:
Implementing this should be just about flipping the ifdef in ecalllist.h around `ThreadNative::Interrupt`. Any reason why you have not just done it?